com.unity.render-pipelines.universal/Editor/Converter/Converters/RenderSettingsConverter.cs

```diff
@@ -0,0 +1,301 @@
+            var path = $"Assets/{m_PipelineAssetPath}/{settings.LevelName}_PipelineAsset.asset";
+
+            // Setting Pipeline Asset settings
+            SetPipelineSettings(asset, settings);
+
+            // Create Renderers
+            var defaultIndex = -1;
+            var renderers = new List<ScriptableRendererData>();
+            if (m_RenderingModes.Contains(RenderingMode.Forward))
+            {
+                renderers.Add(CreateRendererDataAsset(path, RenderingPath.Forward, "ForwardRenderer"));
+                if (GetEquivalentRenderMode(m_GraphicsTierSettings.RenderingPath) == RenderingMode.Forward)
+                    defaultIndex = m_RenderingModes.IndexOf(RenderingMode.Forward);
+            }
+
+            if (m_RenderingModes.Contains(RenderingMode.Deferred))
+            {
+                renderers.Add(CreateRendererDataAsset(path, RenderingPath.DeferredShading, "DeferredRenderer"));
+                if (GetEquivalentRenderMode(m_GraphicsTierSettings.RenderingPath) == RenderingMode.Forward)
+                    defaultIndex = m_RenderingModes.IndexOf(RenderingMode.Deferred);
+            }
+
+            asset.m_RendererDataList = renderers.ToArray();
+            asset.m_DefaultRendererIndex = defaultIndex;
+
+            // Create Pipeline asset on disk
+            AssetDatabase.CreateAsset(asset, path);
+            // Assign asset
+            QualitySettings.SetQualityLevel(settings.Index);
+            QualitySettings.renderPipeline = asset;
+
+            // return to original quality level
+            QualitySettings.SetQualityLevel(currentQualityLevel);
+        }
+
+        private ScriptableRendererData CreateRendererDataAsset(string assetPath, RenderingPath renderingPath,
+            string fileName)
+        {
+            var rendererAsset =
+                UniversalRenderPipelineAsset.CreateRendererAsset(assetPath, RendererType.UniversalRenderer, true, fileName)
+                    as UniversalRendererData;
+            //Missing API to set deferred or forward
+            rendererAsset.renderingMode =
+                renderingPath == RenderingPath.Forward ? RenderingMode.Forward : RenderingMode.Deferred;
+            //missing API to assign to pipeline asset
+            return rendererAsset;
+        }
+
+        /// <summary>
+        /// Sets all relevant RP settings in order they appear in URP
+        /// </summary>
+        /// <param name="asset">Pipeline asset to set</param>
+        /// <param name="settings">The ProjectSettingItem with stored settings</param>
+        private void SetPipelineSettings(UniversalRenderPipelineAsset asset, RenderSettingItem settings)
+        {
+            // General
+            asset.supportsCameraDepthTexture = settings.SoftParticles;
+
+            // Quality
+            asset.supportsHDR = m_GraphicsTierSettings.HDR;
+            asset.msaaSampleCount = settings.MSAA == 0 ? 1 : settings.MSAA;
```

Comment:
How does URP handle this setting? If it is set to 0 in the settings.MSAA we set it to 1. If it is set to 1 in the settings.MSAA we set it to 1 too? Are these settings equivalent?

Reply:
so quality settings use 0, 2, 4, 8 for MSAA while RTs use 1, 2, 4, 8 and URP follows this convention, so all 0 values becomes 1 and the rest should line up as expected.